src/components/App.tsx

```diff
@@ -632,7 +632,7 @@ export class App extends React.Component<AppProps, AppState> {
             this.setState({ newFileDialogDirectory: null });
           }}
           onCreate={(file: File) => {
-            this.project.addFile(file);
+            this.state.newFileDialogDirectory.addFile(file);
```

Comment:
It's coming from previous code, but you cannot mutate state this way. `this.state.uploadFileDialogDirectory` needs to be refactored or PR that introduced it removed from the code.